### ISSUE TITLE
sphinx 1.8.0 :type: can't process :term: references

### ISSUE THREAD

**Issue by reaperhulk** [sphinx-doc/sphinx#5432]
### Problem
As of sphinx 1.8.0 `:term:` references in a `:type:` no longer appear to work.

#### Procedure to reproduce the problem
Use this rst:
```
.. glossary::

    test
        data


.. currentmodule:: foo.bar

.. class:: Baz

    .. attribute:: quux

        :type: :term:`test`
```
Then run sphinx-build against it with a default conf and you'll see: `WARNING: py:class reference target not found: test`

In 1.7.9 and before this would properly link.

#### Environment
* Python 3.7
* macOS 10.13
* sphinx 1.8.0

**Comment by tk0miya** [comment]
So far, `py:attributes` does not handle `:type:` field especially. But, since 1.8, it has been handled as types forcedly. I think #4047 contains a bug like this.
+1 for fixing this.

**Comment by tk0miya** [comment]
Fixed by #5467.
Thank you for reporting.
